lib/instrumentation/@prisma/client.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const connection = Symbol('nrPrismaConnection')
+const url = require('url')
+const logger = require('../../logger').child({ component: 'prisma' })
+const parseSql = require('../../db/query-parsers/sql')
+
+/**
+ * Parses a connection string. Most database engines in prisma are SQL and all
+ * have similar engine strings.
+ *
+ * **Note**: This will not parse mongodb or sql server, instead will log a warning
+ *
+ * @param {string} provider prisma provider(i.e. mysql, postgres, mongodb)
+ * @param {string} connectionUrl connection string
+ * @returns {object} { host, port, database_name }
+ */
+function parseConnectionString(provider, connectionUrl) {
+  let parameters = {}
+  try {
+    const parsedUrl = url.parse(connectionUrl)
+    parameters = {
+      host: parsedUrl.hostname,
+      port: parsedUrl.port,
+      database_name: parsedUrl.path && decodeURIComponent(parsedUrl.path.split('/')[0])
+    }
+  } catch (err) {
+    logger.warn('Failed to parse connection string for %s: %s', provider, err.message)
```

Comment:
Is there a reason to prefer format strings over template literals?

Reply:
not really.  template strings are newer and sometimes I'm just used to doing this when logging